src/main/src/main/java/org/geoserver/catalog/CoverageReaderFileConverter.java

```diff
@@ -0,0 +1,79 @@
+/* (c) 2014 - 2020 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.catalog;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.geoserver.platform.resource.Files;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resources;
+import org.geotools.util.factory.Hints;
+import org.opengis.coverage.grid.GridCoverageReader;
+
+/**
+ * Attempts to convert the source input object for a {@link GridCoverageReader} to {@link File}.
+ *
+ * @author joshfix Created on 2/25/20
+ */
+public class CoverageReaderFileConverter implements CoverageReaderInputObjectConverter<File> {
+
+    private final Catalog catalog;
+
+    public CoverageReaderFileConverter(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    /**
+     * Performs the conversion of the input object to a file object. If this converter is not able
+     * to convert the input to a File, an empty {@link Optional} will be returned.
+     *
+     * @param input The input object.
+     * @param coverageInfo The grid coverage metadata, may be <code>null</code>.
+     * @param hints Hints to use when loading the coverage, may be <code>null</code>.
+     * @return
+     */
+    @Override
+    public Optional<File> convert(
+            Object input, @Nullable CoverageInfo coverageInfo, @Nullable Hints hints) {
+        if (!(input instanceof String)) {
+            return Optional.empty();
+        }
+        String urlString = (String) input;
+        return canConvert(urlString) ? convertFile(urlString) : Optional.empty();
+    }
+
+    /**
+     * Checks to see if the input string is a file URI.
+     *
+     * @param input The input string.
+     * @return Value representing whether or not this converter is able to convert the provided
+     *     input to File.
+     */
+    protected boolean canConvert(String input) {
+        // Check to see if our "url" points to a file or not, otherwise we use the string itself for
+        // reading
+        try {
+            URI uri = new URI(input);
+            return uri.getScheme() != null && "file".equalsIgnoreCase(uri.getScheme());
```

Comment:
There is indeed a behavioral change, the previous code was trying to lookup the files using the resource loader even if the string representation had no scheme, this one does not. The following change fixes the specific build failure.

```suggestion
            return uri.getScheme() == null || "file".equalsIgnoreCase(uri.getScheme());
```

I've verified this change makes for a successful build (with all extensions included, -Prelease):

```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:12 min (Wall Clock)
[INFO] Finished at: 2020-05-06T15:36:28+02:00
[INFO] ------------------------------------------------------------------------
```

Reply:
@joshfix are you still working on this one?